From 98a21c4c121ff45b3033619f909bf9c27302628e Mon Sep 17 00:00:00 2001
From: Haseeb Ashraf <haseeb_ashraf@mentor.com>
Date: Thu, 13 Jun 2024 12:28:33 +0500
Subject: ui/sdl2: fix mouse co-ordinates for scaled guest surface

This is noted that when the guest surface is scaled into a window
other than the default, the mouse co-ordinates are not adjusted
correctly. Properly select the surface dimensions when SDL,GL is
enabled.

This is part of the patch which is merged in S2S-QEMU:
http://orw-medusa-vm.wv.mentorg.com/shyam/s2s-qemu/-/commit/1bbdf069bc1e3a1a53b7e4a22375f5dae90150e6

Upstream-Status: Pending

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>
---
 ui/sdl2.c | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/ui/sdl2.c b/ui/sdl2.c
index 4971963f0..07f18dd79 100644
--- a/ui/sdl2.c
+++ b/ui/sdl2.c
@@ -320,10 +320,15 @@
     }
 
     if (qemu_input_is_absolute(scon->dcl.con)) {
-        qemu_input_queue_abs(scon->dcl.con, INPUT_AXIS_X,
-                             x, 0, surface_width(scon->surface));
-        qemu_input_queue_abs(scon->dcl.con, INPUT_AXIS_Y,
-                             y, 0, surface_height(scon->surface));
+        int scr_w, scr_h;
+        if (scon->opengl) {
+            SDL_GetWindowSize(scon->real_window, &scr_w, &scr_h);
+        } else {
+            scr_w = surface_width(scon->surface);
+            scr_h = surface_height(scon->surface);
+        }
+        qemu_input_queue_abs(scon->dcl.con, INPUT_AXIS_X, x, 0, scr_w);
+        qemu_input_queue_abs(scon->dcl.con, INPUT_AXIS_Y, y, 0, scr_h);
     } else {
         if (guest_cursor) {
             x -= guest_x;
-- 
2.34.1

